app/controllers/application_controller.rb

```diff
@@ -3,19 +3,26 @@ class ApplicationController < ActionController::Base
 
   helper_method :logged_in?, :current_user # Give views access to these application controller methods.
 
+  def require_user
+    if !logged_in?
+      flash[:error] = "You must be logged in to do that."
+      redirect_to front_path
+    end
+  end
+
   # Returns a user object or nil.
   def current_user
     if logged_in?
-      User.find(session[:id])
+      User.find(session[:user_id])
     end
   end
 
   # Checks whether the user is logged into the app. Returns a bool.
   def logged_in?
-    session[:id] ? true : false
+    session[:user_id] ? true : false
```

Comment:
No need for the ternary, you are basically saying "if true, then true...otherwise false".